docs/ROADMAP.md

```diff
@@ -47,11 +47,49 @@ concepts.
 ### 6. Function catalog
 
 All of the hydration and validation logic has been moved from the kpt binary 
-to functions allowing for flexibility and security.  This enables new 
+to functions allowing for flexibility and security. This enables new 
 scenarios like limiting the customization and validation to a subset of 
 allowed functions.  The function catalog has received additional functions, 
 examples and help. Please visit the [function catalog] for further information.
 
+## In Progress
+
+### 7. Resource filtering in `kpt fn` commands
+
+Users want to invoke a kpt function (imperatively and declaratively) on a subset of 
+resources in the package by selecting them on the basis of GVKNN, package-path, 
+file path etc. For example, set work-load identity annotation on all Kubernetes 
+Service Account resources in this package. [Tracker](https://github.com/GoogleContainerTools/kpt/issues/2015)
+
+### 8. Integrate kpt with Cloud Code
+
+One of the major areas of investment is to integrate Cloud Code with kpt to provide 
+package authoring assistance. Users can author Kptfile and functionConfig files with
+features like auto-complete and error detection. This significantly improves the 
+discoverability of Kptfile schema, catalog functions and their functionConfigs.
+
+### 9. Pre/Post processing step in `kpt fn render`
+
+We need a way to pre(/post) process inputs(/outputs) involved in customizing the 
+package or any of its sub-packages before(/after) invoking the function pipeline [Tracker](https://github.com/GoogleContainerTools/kpt/issues/2419).
+e.g. Use-case to propagate common setters values in the package hierarchy before
+rendering the package resources. 
+
+### 10. Improve functionConfig UX
+
+Package publishers want to provide command driven instructions (semi-automated)
+to configure a package that makes the user-guide/examples easy to follow.
+e.g. Porcelain to set values in functionConfig or generate functionConfig
+
+Package consumers should be able to detect configuration errors early before the
+package is assumed to be ready. e.g. Introduce validation for functionConfig.
+
+### 11. Improve Function Authoring Experience
+
+We need a rich ecosystem of third party functions. Users should be able to write 
+functions with custom logic very quickly using the tools they are familiar with. 
+So we are investing on making function authoring experience very easy.
```

Comment:
We can capture `starlark` enhancements (and docs improvement) to be addressed in Q3.

Reply:
Done.